hikari/internal/time.py

```diff
@@ -98,9 +98,14 @@ def iso8601_datetime_string_to_datetime(datetime_str: str) -> datetime.datetime:
     # Discord appears to actually use RFC-3339, which isn't a true ISO-8601 implementation,
     # but somewhat of a subset with some edge cases.
     # See https://tools.ietf.org/html/rfc3339#section-5.6
-    iso8601_datetime_string_to_datetime = ciso8601.parse_rfc3339  # noqa: F811 - Redefined function
+    fast_iso8601_datetime_string_to_datetime = ciso8601.parse_rfc3339
+
 except ImportError:
-    pass
+    fast_iso8601_datetime_string_to_datetime = None
+
+iso8601_datetime_string_to_datetime = (
```

Comment:
What does mypy resolve the typing for this variable to out of curiosity?

Reply:
```
hikari/internal/time.py:111:13: note: Revealed type is 'Union[Any, def (datetime_str: builtins.str) -> datetime.datetime]'
```